src/main/java/com/github/dockerjava/api/model/Ports.java

```diff
@@ -134,28 +133,28 @@ public static Binding binding(Integer hostPort) {
 
         private final String hostIp;
 
-        private final Integer hostPort;
+        private final String hostPort;
 
         /**
          * Creates a {@link Binding} for the given {@link #getHostIp() IP address} and {@link #getHostPort() port number}.
          *
          * @see Ports#bind(ExposedPort, Binding)
          * @see ExposedPort
          */
-        public Binding(String hostIp, Integer hostPort) {
+        public Binding(String hostIp, String hostPort) {
             this.hostIp = isEmpty(hostIp) ? null : hostIp;
             this.hostPort = hostPort;
         }
 
         /**
-         * Creates a {@link Binding} for the given {@link #getHostPort() port number}, leaving the {@link #getHostIp() IP address}
+         * Creates a {@link Binding} for the given {@link #getHostPort() port number or range}, leaving the {@link #getHostIp() IP address}
          * undefined.
          *
          * @see Ports#bind(ExposedPort, Binding)
          * @see ExposedPort
          */
         public Binding(Integer hostPort) {
```

Comment:
The thing is there already is a constructor same signature, that's why I didn't change the Integer. I guess it would be safer to have static factories instead, and only keep the constructor wil everything. What do you think ?